contrib/dracut/90zfs/zfs-generator.sh.in

```diff
@@ -15,8 +15,10 @@ type getarg >/dev/null 2>&1 || . "$dracutlib"
 [ "${root##zfs:}" = "${root}" -a "${root##ZFS=}" = "${root}" -a "$rootfstype" != "zfs" ] && exit 0
 
 rootfstype=zfs
-if echo "${rootflags}" | grep -q zfsutil ; then
+if echo "${rootflags}" | grep -Eq '^zfsutil$|^zfsutil,|,zfsutil$|,zfsutil,' ; then
     true
+elif test -n "${rootflags}" ; then
+    rootflags="zfsutil,${rootflags}"
 else
     rootflags=zfsutil
 fi
```

Comment:
Not specific to this change, but you could add a few lines here to handle the zfs:AUTO case, where no root was specified, like in mount-zfs.sh.  I assume we don't need to do the check twice, since the pool should already be imported.  Something like this:

```
if [ "${root}" = "zfs:AUTO" ] ; then
  root="$(find_bootfs)"
  if [ $? -ne 0 ] ; then
    exit 0
  fi
fi
```